research-domain-agent/src/test/java/care/smith/fts/rda/rest/TransferProcessControllerIT.java

```diff
@@ -0,0 +1,135 @@
+package care.smith.fts.rda.rest;
+
+import static care.smith.fts.util.MediaTypes.APPLICATION_FHIR_JSON;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.InstanceOfAssertFactories.type;
+import static org.springframework.boot.test.context.SpringBootTest.WebEnvironment.RANDOM_PORT;
+import static org.springframework.http.HttpStatus.NOT_FOUND;
+import static reactor.test.StepVerifier.create;
+
+import care.smith.fts.rda.ResearchDomainAgent;
+import care.smith.fts.rda.TransferProcessRunner.Status;
+import care.smith.fts.test.TestWebClientFactory;
+import java.io.IOException;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.server.LocalServerPort;
+import org.springframework.context.annotation.Import;
+import org.springframework.core.codec.DecodingException;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ProblemDetail;
+import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.reactive.function.client.WebClientResponseException;
+import org.springframework.web.reactive.function.client.WebClientResponseException.NotFound;
+import org.springframework.web.reactive.function.client.WebClientResponseException.UnsupportedMediaType;
+import reactor.core.publisher.Mono;
+
+@Slf4j
+@SpringBootTest(classes = ResearchDomainAgent.class, webEnvironment = RANDOM_PORT)
+@Import(TestWebClientFactory.class)
+public class TransferProcessControllerIT {
+
+  private WebClient client;
+
+  @BeforeEach
+  final void setUpTransferProcessControllerIT(
+      @LocalServerPort int port, @Autowired TestWebClientFactory factory) {
+    client = factory.webClient("https://localhost:" + port);
+  }
+
+  @Test
+  void startProcessWithInvalidProject() {
+    create(
+            client
+                .post()
+                .uri("/api/v2/process/non-existent/patient")
+                .headers(h -> h.setContentType(APPLICATION_FHIR_JSON))
+                .retrieve()
+                .onStatus(
+                    r -> r.equals(HttpStatus.resolve(404)),
+                    (c) ->
+                        c.bodyToMono(ProblemDetail.class)
+                            .flatMap(p -> Mono.error(new IllegalStateException(p.getDetail()))))
+                .toBodilessEntity())
+        .expectErrorMessage("Project 'non-existent' could not be found")
+        .verifyThenAssertThat()
+        .hasOperatorErrors();
+  }
```

Comment:
This looks very complicated, wrapping the problem in a ISE exception and such. Can we simplify this?